src/Bridge/Elasticsearch/DataProvider/PaginatorFactory.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Elasticsearch\DataProvider;
+
+use ApiPlatform\Core\DataProvider\PaginatorFactoryInterface;
+use ApiPlatform\Core\DataProvider\PaginatorInterface;
+use ApiPlatform\Core\Exception\RuntimeException;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+
+final class PaginatorFactory implements PaginatorFactoryInterface
+{
+    private $denormalizer;
+
+    public function __construct(DenormalizerInterface $denormalizer)
+    {
+        $this->denormalizer = $denormalizer;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createPaginator($subject, int $limit, int $offset, array $context = []): PaginatorInterface
+    {
+        $resourceClass = $context['resourceClass'] ?? null;
```

Comment:
Is there a specific reason for passing the resource class in the context rather than in a parameter of the method?

Reply:
Mostly saving one argument which doesn't trully make sense in the context of a paginator.